Drivers/Src/MotorController.c

```diff
@@ -89,36 +157,44 @@ ErrorStatus MotorController_Read(CANbuff *message, car_state_t *car){
             // If we're reading the output from the Motor Status command (0x241) then 
             // Check the status bits we care about and set flags accordingly
             case MOTOR_STATUS: {
-                if(MASK_MOTOR_TEMP_ERR & firstSum)
-                {
-                    car->MotorErrorCode.motorTempErr = ON;
+                if(MASK_MOTOR_TEMP_ERR & firstSum){
+                    Motor_FaultBitmap |= T_TEMP_ERR;
                 }
 
-                if(MASK_SS_ERR & firstSum)
-                {
-                    car->MotorErrorCode.slipSpeedErr = ON;
+                if(MASK_SS_ERR & firstSum){
+                    Motor_FaultBitmap |= T_CC_VEL_ERR;
                 }
 
-                if(MASK_CC_ERR & firstSum)
-                {
-                    car->MotorErrorCode.CCVelocityErr = ON;
+                if(MASK_CC_ERR & firstSum){
+                    Motor_FaultBitmap |= T_SLIP_SPEED_ERR;
                 }
 
-                if(MASK_OVER_SPEED_ERR & firstSum)
-                {
-                    car->MotorErrorCode.overSpeedErr = ON;
+                if(MASK_OVER_SPEED_ERR & firstSum){
+                    Motor_FaultBitmap |= T_OVER_SPEED_ERR;
                 }
+
+                assertTritiumError(Motor_FaultBitmap);
```

Comment:
This part technically isn't thread-safe, since `Motor_FaultBitmap` is a global and you are modifying it non-atomically. I don't think it matters though unless you call this function from multiple threads and care about distinguishing between different motor errors (neither of which I think y'all are doing)

Reply:
I think it does matter because we have different behaviors in our fault thread for different motor faults, so I think it does matter, because we don't want a non-critical error to beat a critical error to the fault thread.